crates/starknet/src/starknet/mod.rs

```diff
@@ -1184,4 +1187,29 @@ mod tests {
 
         assert_eq!(latest_block.unwrap().block_number(), BlockNumber(2));
     }
+    #[test]
+    fn check_timestamp_of_newly_generated_block() {
+        let config = starknet_config_for_test();
+        let mut starknet = Starknet::new(&config).unwrap();
+
+        Starknet::update_block_context(&mut starknet.block_context);
+        starknet.generate_pending_block().unwrap();
+        starknet.blocks.pending_block.set_timestamp(BlockTimestamp(
+            std::time::SystemTime::now().duration_since(std::time::UNIX_EPOCH).unwrap().as_secs(),
```

Comment:
Since the same code for getting current time is used in multiple places, why not extract it? Maybe also use the same function above in L182?

Reply:
extracted in [9478662](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/235/commits/947866291971f3ce316e3cf2b03cfc068bb9f032)